### PR TITLE
feat: training bionemo model:esm2 on bionemo framework

### PR DESCRIPTION
**Pull request by sujituk**
- Training ESM2 on BioNeMo framework

### CODE REVIEW

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+- [Prerequisites](#prerequisites)
+- [Setup](#setup)
+- [Cleanup](#cleanup)
+
+## Prerequisites
+
+- **GCloud SDK:** Ensure you have the Google Cloud SDK installed and configured.
+- **Project:**  A Google Cloud project with billing enabled.
+- **Permissions:**  Sufficient permissions to create GKE clusters and other related resources.
```
Comment (roberthbailey) [inline tutorials-and-examples/nvidia-bionemo/esm2/README.md:15]: Also an NGC key like in https://github.com/GoogleCloudPlatform/ai-on-gke/blob/42b358fef504c60872045c5d3ce154bcd2d20b9b/tutorials-and-examples/nvidia-nim/blueprints/drugdiscovery/README.md?

Comment (roberthbailey) [reply tutorials-and-examples/nvidia-bionemo/esm2/README.md:15]: The other guide doesn't mention permissions, but in addition to the NGC API key it lists kubectl.

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+6. Create and mount Google cloud Filestore for storage
+
+```bash
+alias k=kubectl
+
+k apply -f create-mount-fs.yaml
+```
+
+Wait for the filestore to be in READY status. It takes a few minutes.
```
Comment (roberthbailey) [inline tutorials-and-examples/nvidia-bionemo/esm2/README.md:84]: How do we know when it's ready? Is there a command I should run at this point to verify I can move on?

Comment (roberthbailey) [reply tutorials-and-examples/nvidia-bionemo/esm2/README.md:84]: Also, this sentence can easily get lost. I'd make it step 7 instead of just free floating text.

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+```bash
+k apply -f esm2-training.yaml
+```
+
+8. Port Forwarding (for TensorBoard):
+
+```bash
+POD_BIONEMO_TRAINING=$(k get pods -o go-template --template '{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}' | grep '^bionemo-training')
+
+k port-forward pod/$POD_BIONEMO_TRAINING :6006
```
Comment (roberthbailey) [inline tutorials-and-examples/nvidia-bionemo/esm2/README.md:97]: As in the other guide, I think you should pick a local port for the port forwarding command.

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md (hunk 1 above)
Comment (sujituk) [reply tutorials-and-examples/nvidia-bionemo/esm2/README.md:15]: ack. updated this, added git clone step

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md (hunk 2 above)
Comment (sujituk) [reply tutorials-and-examples/nvidia-bionemo/esm2/README.md]: ack. added command to verify when its bound.

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md (hunk 3 above)
Comment (sujituk) [reply tutorials-and-examples/nvidia-bionemo/esm2/README.md]: ack updated.

**Comment by sujituk** [comment]
@roberthbailey comments addressed. ready for review

**Comment by sujituk** [comment]
@roberthbailey added support for fine-tuning as well. updated the readme and add 2 files.

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+10. View Tensorboard logs
+
+On your local machine: Browse to <<http://localhost:8000/#timeseries> and see the loss curves as show below
+
+[<img src="images/tensorboard-results.png" width="750"/>](HighLevelArch)
+
+## Fine-tuning
+
+11. Start the fine-tuning job. The results will be written to a .pt file
```
Comment (roberthbailey) [inline tutorials-and-examples/nvidia-bionemo/esm2/README.md:139]: Is it expected that the person running this knows what to do with the .pt file?

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+```bash
+k apply -f esm2-finetuning.yaml
```
Comment (roberthbailey) [inline tutorials-and-examples/nvidia-bionemo/esm2/README.md:142]: With the default cluster settings (1 node w/ gpu) this pod was unschedulable for me. I deleted the job created in the training step to free up resources.

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+k get pods
+k exec -it <pod-name> -- bash
+
+#Copy contents of finetuning.py into the pod
+touch finetuning.py
+vi finetuning.py
+#paste contents into the file
+python3 finetuning.py
```
Comment (roberthbailey) [inline tutorials-and-examples/nvidia-bionemo/esm2/README.md:150]: If I want to save the generated .pt file, how do I extract it from the pod and/or the filestore?

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+```
+
+## Cleanup
+
+To delete the cluster and all associated resources:
+
+```bash
+k delete -f esm2-finetuning.yaml
+k delete -f esm2-pretraining.yaml
+k delete -f create-mount-fs.yaml
```
Comment (roberthbailey) [inline tutorials-and-examples/nvidia-bionemo/esm2/README.md:160]: Do I need to wait a while after deleting the filestore before deleting the cluster to make sure it cleans up correctly?

File: tutorials-and-examples/nvidia-bionemo/esm2/esm2-finetuning.yaml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+kind: Job
+metadata:
+  name: bionemo-finetuning
+spec:
+  template:
+    spec:
+      containers:
+      - name: finetuning
+        # image: nvcr.io/nvidia/clara/bionemo-framework:2.1
+        image: nvcr.io/nvidia/clara/bionemo-framework:nightly
```
[the remark this answers is not in the document]
Comment (sujituk) [reply tutorials-and-examples/nvidia-bionemo/esm2/esm2-finetuning.yaml]: > How do I know when it's finished? Will the job complete? It's still running and I see all of the complete graphs as shown below, but I don't know how long I need to wait before I proceed to the fine-tuning step.

The pod logs indicate when the job with steps (100) and iterations are completed. Provided estimated time in readme. 

> Looking at the yaml for this pod, it looks like it sleeps forever, so does that mean once the tensorboard is available that I can move along?

added step to delete job after viewing logs

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md (hunk 4 above)
Comment (sujituk) [reply tutorials-and-examples/nvidia-bionemo/esm2/README.md]: yes, it's loaded in the script to view the results

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md (hunk 5 above)
Comment (sujituk) [reply tutorials-and-examples/nvidia-bionemo/esm2/README.md]: added step to delete the job. use unable to view logs if exited.

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md (hunk 7 above)
Comment (sujituk) [reply tutorials-and-examples/nvidia-bionemo/esm2/README.md]: should be okay.

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+```bash
+k get persistentvolumeclaim fileserver -o yaml | grep phase:
+
+```
+
+The output should show `phase: Bound` when the Filestore instance is ready.
+
+## Pretraining
+
+8. Kickoff the pretraining job. The job will automatically create ./results and store the checkpoints under esm2 in the Filestore mounted earlier under `/mnt/data`.
```
[the remark this answers is not in the document]
Comment (sujituk) [reply tutorials-and-examples/nvidia-bionemo/esm2/README.md]: removed hyperlink altogether to reduce confusion.

File: tutorials-and-examples/nvidia-bionemo/esm2/README.md [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+```bash
+POD_BIONEMO_TRAINING=$(k get pods -o go-template --template '{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}' | grep '^bionemo-training')
+
+k port-forward pod/$POD_BIONEMO_TRAINING 8000:6006
+
+```
+
+10. View Tensorboard logs
+
+On your local machine: Browse to <<http://localhost:8000/#timeseries> and see the loss curves as show below
```
[the remark this answers is not in the document]
Comment (sujituk) [reply tutorials-and-examples/nvidia-bionemo/esm2/README.md]: we could do this in parallel where logs are available while its training. for now, i've updated verbiage so its done in sequence.

[the remark this answers is not in the document]
Comment (sujituk) [reply tutorials-and-examples/nvidia-bionemo/esm2/README.md]: added.

**Comment by sujituk** [comment]
@roberthbailey comments addressed.

**Review by roberthbailey — APPROVED** [review]
